File: server/models/group.go
package models

type Group struct {
	ID        string
	Name      string
	CreatedAt int64
}

type GroupResponse struct {
	ID        string `json:"id"`
	Name      string `json:"name"`
	CreatedAt int64  `json:"created_at"`
}

type GroupMember struct {
	GroupID  string
	UserID   string
	JoinedAt int64
}

type GroupMemberResponse struct {
	UserID   string `json:"user_id"`
	JoinedAt int64  `json:"joined_at"`
}

type CreateGroupRequest struct {
	GroupID string `json:"group_id"`
	Name    string `json:"name"`
}

type RenameGroupRequest struct {
	Name string `json:"name"`
}

type AddMemberRequest struct {
	UserID string `json:"user_id"`
}

type GroupProfilePicture struct {
	GroupID   string
	ImageData []byte
	Version   int64
	UpdatedAt int64
}

type GroupProfilePictureRequest struct {
	ImageData string `json:"image_data"`
	Version   int64  `json:"version"`
}

type GroupProfilePictureResponse struct {
	ImageData string `json:"image_data"`
	Version   int64  `json:"version"`
}

File: client/src/types/group.ts
export interface GroupResponse {
  id: string;
  name: string;
  created_at: number;
}

export interface GroupMemberResponse {
  user_id: string;
  joined_at: number;
}

export interface GroupInviteResponse {
  group_id: string;
  group_name: string;
  user_id: string;
  invited_by: string;
  created_at: number;
}

export interface Group {
  id: string;
  name: string;
  members: string[];
}

export interface GroupProfilePictureResponse {
  image_data: string;
  version: number;
}
